### PR TITLE
Feature chat

### PR DESCRIPTION
**Pull request by aleechan**
Added the visual layout and front-end functions for a basic chat. The user information sidebar in the chat can be implemented when we figure out what will go there.

### CODE REVIEW

File: client/react/components/chat/channelBar.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React, {Component} from 'react';
+import { connect } from 'react-redux';
+import { push } from 'react-router-redux';
+import { Container,Icon, Menu} from 'semantic-ui-react';
+
+import './styles.css';
+
+const listChannels = ({
```
Comment (JoeRoussy) [inline client/react/components/chat/channelBar.jsx]: Put this inside the main render method we we don't need to pass in all these variables.

We are going to try and have logic in the render method before the return.

This means the render method can't have an implicit return.

File: client/react/components/chat/chatView.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { push } from 'react-router-redux';
+
+
+import './styles.css';
+
+const ListComments = ({
+    chatLog
+}) => {
+        return chatLog.map((message,i) => (
+        <Comment key= {i}>
```
Comment (JoeRoussy) [inline client/react/components/chat/chatView.jsx]: Indent this 1 level

File: client/react/components/chat/chatView.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            <Comment.Content>
+                <Comment.Author as='a'>{message.userName}</Comment.Author>
```
Comment (JoeRoussy) [inline client/react/components/chat/chatView.jsx]: We should implement a 'go to profile'.

Something like a link to /profile/{id}

File: client/redux/reducers/chatReducer.jsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+];
+
+const ChatReducer = (state = config, actions) => {
+    switch(actions.type) {
+        case 'GET_CHANNELS': {
+            state = {...state}
+            break;
+        }
+        case 'SET_ACTIVE_CHANNEL':{
+            state = {...state,activeChannel: actions.payload.channel.name}
```
Comment (JoeRoussy) [inline client/redux/reducers/chatReducer.jsx]: Break into different lines to it is easier to read:

```
state = {
    ...state,
    activeChannel: actions.payload.channel.name
}
```

File: client/redux/reducers/chatReducer.jsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            break;
+        }
+        case 'LOAD_ACTIVE_CHANNEL':{
+            //load test chat data
+            switch (state.activeChannel){
+                case 'Channel 1':{
+                    state = {...state, chatLog: Channel1Chat};
```
Comment (JoeRoussy) [inline client/redux/reducers/chatReducer.jsx]: Same as above for all these examples below

**Review by JoeRoussy — CHANGES_REQUESTED** [review]

**Review by JugesPatel — COMMENTED** [review]
Overall it looks good. Agree with comments that Joe made.

**Review by JoeRoussy — APPROVED** [review]

**Pull request merged**